File: slack-poster/src/test/java/dev/number6/slackposter/dagger/FakeSlackPosterConfigurationModule.java
package dev.number6.slackposter.dagger;

import dev.number6.slackposter.port.SlackPosterConfigurationPort;
import dagger.Module;
import dagger.Provides;

@Module
public class FakeSlackPosterConfigurationModule {

    public static final String SLACK_POST_MESSAGE_URL = "http://post.message.to/slack";
    public static final String POSTING_CHANNEL_ID = "POSTING_CHANNEL_ID";

    @Provides
    public SlackPosterConfigurationPort configPort() {
        return new SlackPosterConfigurationPort() {
            @Override
            public String getPostingChannelId() {
                return POSTING_CHANNEL_ID;
            }

            @Override
            public String getSlackPostMessageUrl() {
                return SLACK_POST_MESSAGE_URL;
            }
        };
    }

}

File: slack-poster/src/test/kotlin/dev/number6/slackposter/dagger/FakeSlackPosterConfigurationModule.kt
package dev.number6.slackposter.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slackposter.port.SlackPosterConfigurationPort

@Module
class FakeSlackPosterConfigurationModule {
    @Provides
    fun configPort(): SlackPosterConfigurationPort {
        return object : SlackPosterConfigurationPort {
            override val postingChannelId: String
                get() = POSTING_CHANNEL_ID
        }
    }

    companion object {
        const val POSTING_CHANNEL_ID = "POSTING_CHANNEL_ID"
    }
}